src/ControlSystem/WriteData.hpp

```diff
@@ -0,0 +1,174 @@
+      // Time
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      // Lambda, dtLambda, d2tLambda
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      // Q, dtQ, control signal
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>>;
+
+  using observed_reduction_data_tags =
+      observers::make_reduction_data_tags<tmpl::list<ReductionData>>;
+
+  const static inline std::vector<std::string> legend{
+      "Time", "Lambda", "dtLambda", "d2tLambda", "Q", "dtQ", "Control Signal"};
+};
+
+template <typename ControlSystem>
+struct Registration {
+  template <typename ParallelComponent, typename DbTagsList,
+            typename ArrayIndex>
+  static std::pair<observers::TypeOfObservation, observers::ObservationKey>
+  register_info(const db::DataBox<DbTagsList>& /*box*/,
+                const ArrayIndex& /*array_index*/) {
+    return {observers::TypeOfObservation::Reduction,
+            observers::ObservationKey{ControlSystem::name()}};
+  }
+};
+}  // namespace detail
+
+/*!
+ * \ingroup ControlSystemGroup
+ * \brief Writes data at a specific time to disk from a control system after it
+ * updates its functions of time.
+ *
+ * \details The columns of data written are:
+ * - %Time
+ * - Lambda
+ * - dtLambda
+ * - d2tLambda
+ * - Q
+ * - dtQ
+ * - Control Signal
+ *
+ * where "Lambda" is the function of time value and "Q" is the control error.
+ *
+ * Data will be stored in the reduction file. All subfiles for the control
+ * system within the H5 file will be under the group "/ControlSystems".
+ * Within this group, there will be one group for each control system. The name
+ * of each group will be the result of the `name()` function from each control
+ * system. An example would look like
+ *
+ * - /ControlSystems/SystemA
+ * - /ControlSystems/SystemB
+ * - /ControlSystems/SystemC
+ *
+ * Then, within each system group, there will be one subfile for each component
+ * of the function of time that is being controlled. The name of this subfile is
+ * the name of the component appended to the name of the control system. The
+ * name of each component will be the result of the `component_name(i)` function
+ * from the control system, where `i` is the index of the component. For
+ * example, if "SystemA" has 3 components with names "X", "Y", and "Z", then the
+ * subfiles would look like
+ *
+ * - /ControlSystems/SystemA/SystemAX.dat
+ * - /ControlSystems/SystemA/SystemAY.dat
+ * - /ControlSystems/SystemA/SystemAZ.dat
+ *
+ * \note In order to use this function, you must 1) put the
+ * `RegisterSingletonWithObserverWriter` action in the registration phase of the
+ * `ControlComponent` and 2) add `control_system::detail::WriterHelper` to the
```

Comment:
No referencing `detail` stuff externally.  If someone needs to care about it it's not a detail.

Reply:
Ok I just removed the `detail` namespace